lua/neogit/lib/git/branch.lua

```diff
@@ -32,6 +32,19 @@ local function get_all_branches()
   return parse_branches(branches)
 end
 
+local function get_upstream()
+  local upstream_text = cli.branch.tracking.call()
+
+  for _, branch in ipairs(upstream_text) do
+    local ups_remote, ups_branch = branch:match([[^%*.+%[(.+)/([^:]+):?.*%].+]])
```

Comment:
This pattern seems to make a lot of wrong assumptions about the formatting of both remotes and branches. Both remotes and branches may contain slashes (`/`). Take for instance this example where the remote is called `foo/bar/baz`, and the branch is called `lorem/ipsum/dolor`. Using the current approach, this will be parsed as:

```lua
{
  remote = "foo/bar/baz/lorem/ipsum",
  branch = "dolor"
}
```

Reply:
Oh, I forgot those edge case.